dipy/align/streamlinear.py

```diff
@@ -411,8 +475,16 @@ def optimize(self, static, moving, mat=None):
                                             compose_matrix44(vecs),
                                             static_mat))
 
-        srm = StreamlineRegistrationMap(mat, opt.xopt, opt.fopt,
-                                        mat_history, opt.nfev, opt.nit)
+        # If we are running halfway streamline linear registration (for
+        # groupwise registration or atlasing) the registration map is different
+        if isinstance(self.metric, JointBundleMinDistanceMetric):
+            srm = JointStreamlineRegistrationMap(opt.xopt, opt.fopt,
+                                                 mat_history, opt.nfev,
+                                                 opt.nit)
+        else:
+            srm = StreamlineRegistrationMap(mat, opt.xopt, opt.fopt,
+                                            mat_history, opt.nfev, opt.nit)
```

Comment:
we might need to reshape this design in a close future, that's fine for now

Reply:
true.., I didn't come up with anything better without duplicating a lot of code